test/get_previous_releases.py

```diff
@@ -133,20 +134,22 @@ def download_binary(tag, args) -> int:
 
     print('Fetching: {tarballUrl}'.format(tarballUrl=tarballUrl))
 
-    header, status = subprocess.Popen(
-        ['curl', '--head', tarballUrl], stdout=subprocess.PIPE).communicate()
-    if re.search("404 Not Found", header.decode("utf-8")):
-        print("Binary tag was not found")
-        return 1
-
-    curlCmds = [
-        ['curl', '--remote-name', tarballUrl]
-    ]
+    try:
+        response = urlopen(Request(tarballUrl, method='HEAD'))
+    except HTTPError as e:
+        if e.code == 404:
+            print("Binary tag was not found")
+            return 1
+        else:
+            raise
 
-    for cmd in curlCmds:
-        ret = subprocess.run(cmd).returncode
-        if ret:
-            return ret
+    try:
+        response = urlopen(tarballUrl)
+        with open(tarball, 'wb') as file:
+            file.write(response.read())
```

Comment:
You can `response.read(amt=<number of bytes>)` so it could be chunked to avoid loading the whole thing in to memory. That *could* also be used to add back some coarse progress indicator